### DISCUSSION TITLE
Categories issues

### DISCUSSION THREAD

**Discussion by davidovic999** [mikeberger/borg_calendar#206] [General]
Hi Mike,
Categories present a few issues:

1/ The list of categories created is not kept after the restart. It appears that the list has been cleared.
see the screen record: [https://ufile.io/jo8i0w5b](url)

It appears that the program automatically deletes unused categories.
It's better to keep the whole list as it's created and be able to manage it ourselves.

2/ Also, this list of categories is not saved. When you view the contents of the backup zip file, you see that there is no "Categories" file.

**Comment by mikeberger** [comment]
The categories feature was added over 20 years ago: https://sourceforge.net/p/borg-calendar/news/2004/04/borg-release-13-beta-2/

It is working as designed, but was designed badly. There were lots of comments like yours, for example: https://sourceforge.net/p/borg-calendar/discussion/323020/thread/b64bc3af/

I don't fully remember why it was done the way it was, but I think it was because BORG used a NoSQL database that I wrote in the days before databases like H2 or HSQL and it caused issues of some sort. I didn't use categories myself and the feature remained unchanged.

When the program starts, the list of categories is built from the categories that are in use in the appointments and tasks. The categories are not stored separately and categories with no appointments or tasks disappear because they are not stored.

The used categories are in the backup file inside the appointment and task records.

**Comment by davidovic999** [comment]
Nice to know that.
Cheers!
